app/src/main/java/grinnell/appdev/edu/lyles/AsyncRetrieval.java

```diff
@@ -0,0 +1,71 @@
+package grinnell.appdev.edu.lyles;
+
+import android.os.AsyncTask;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * A class which asynchronously fetches and parses JSON files
+ *
+ * @author Shelby Frazier
+ */
+
+public class AsyncRetrieval extends AsyncTask<Void, Void, String> {
+
+    private final String mUrl;
+    private String mJsonBody;
+
+
+    public AsyncRetrieval(String url) {
+        super();
+        mUrl = url;
+    }
+
+    @Override
+    protected void onPreExecute() {
+        super.onPreExecute();
+    }
+
+    @Override
+    protected String doInBackground(Void... voids) {
+        final OkHttpClient client = new OkHttpClient();
+        final Request request = new Request.Builder().url(mUrl).build();
+
+        try {
+            final Response response = client.newCall(request).execute();
+            mJsonBody = response.body().string();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return mJsonBody;
+    }
+
+    @Override
+    protected void onPostExecute(String aString) {
+        super.onPostExecute(mJsonBody);
+    }
+
+    public JSONArray getJsonArray(String jsonBody, String arrayTitle) {
```

Comment:
You should document this method (rule of thumb: Unless you inherited the method from another class, you should document methods you write) .